hazelcast-sql/src/test/java/com/hazelcast/sql/impl/expression/math/PlusTest.java

```diff
@@ -0,0 +1,187 @@
+        assertRows(query(sql, "select __key, double1 + ? from records where __key < 1000", "2.1"), keyRange(0, 1000),
+                k -> (k + 2000.1) + 2.1);
+        assertRows(query(sql, "select __key, double1 + null from records"), keyRange(0, 1000, 5000, 6000), k -> null);
+        assertQueryThrows(sql, "select ? + int1 from records", "bigint overflow", Long.MAX_VALUE);
+    }
+
+    @Test
+    public void verify() {
+        verify(PLUS, PlusTest::expectedTypes, PlusTest::expectedValues, ALL, ALL);
+    }
+
+    @Test
+    public void testCreationAndEval() {
+        PlusFunction<?> expression =
+                PlusFunction.create(ConstantExpression.create(3, INT), ConstantExpression.create(2, INT), INT);
+        assertEquals(INT, expression.getType());
+        assertEquals(5, expression.eval(row("foo"), SimpleExpressionEvalContext.create()));
+    }
+
+    @Test
+    public void testEquality() {
+        checkEquals(PlusFunction.create(ConstantExpression.create(3, INT), ConstantExpression.create(2, INT), INT),
+                PlusFunction.create(ConstantExpression.create(3, INT), ConstantExpression.create(2, INT), INT), true);
+
+        checkEquals(PlusFunction.create(ConstantExpression.create(3, INT), ConstantExpression.create(2, INT), INT),
+                PlusFunction.create(ConstantExpression.create(3, INT), ConstantExpression.create(2, INT), BIGINT), false);
+
+        checkEquals(PlusFunction.create(ConstantExpression.create(3, INT), ConstantExpression.create(2, INT), INT),
+                PlusFunction.create(ConstantExpression.create(3, INT), ConstantExpression.create(100, INT), INT), false);
+    }
+
+    @Test
+    public void testSerialization() {
+        PlusFunction<?> original = PlusFunction.create(ConstantExpression.create(3, INT), ConstantExpression.create(2, INT), INT);
+        PlusFunction<?> restored = serializeAndCheck(original, SqlDataSerializerHook.EXPRESSION_PLUS);
+
+        checkEquals(original, restored, true);
+    }
+
+    private static RelDataType[] expectedTypes(Operand[] operands) {
+        // Infer types.
+
+        RelDataType[] types = inferTypes(operands, true);
+        if (types == null) {
+            return null;
+        }
+        RelDataType commonType = types[2];
+
+        // Validate, coerce and infer return type.
+
+        if (!isNumeric(commonType)) {
+            return null;
+        }
+
+        for (int i = 0; i < types.length - 1; ++i) {
+            Operand operand = operands[i];
+            RelDataType type = types[i];
+
+            if (!isNumeric(type) && !isNull(type)) {
+                return null;
+            }
+
+            if (operand.isLiteral() && !canRepresentLiteral(operand, type)) {
+                return null;
+            }
+        }
+
+        if (isInteger(commonType)) {
+            types[2] = HazelcastReturnTypes.binaryIntegerPlus(types[0], types[1]);
```

Comment:
This line is a good example of why such test automation is dangerous. I can break `binaryIntegerPlus` function, e.g. by adding one additional increment to the width, and all tests in this class will pass still, even though the produced behavior is wrong.

Reply:
Resolving, because it duplicates the previous discussions on how to test the expressions in general.